lib/galaxy/jobs/runners/util/job_script/DEFAULT_JOB_FILE_TEMPLATE.sh

```diff
@@ -1,25 +1,32 @@
 #!$shell
 
+_galaxy_setup_environment() {
+    local _use_framework_galaxy="$1"
+    if [ "$GALAXY_LIB" != "None" -a "$_use_framework_galaxy" = "True" ]; then
+        if [ -n "$PYTHONPATH" ]; then
+            PYTHONPATH="$GALAXY_LIB:$PYTHONPATH"
+        else
+            PYTHONPATH="$GALAXY_LIB"
+        fi
+        export PYTHONPATH
+    fi
+    $env_setup_commands
+    if [ "$GALAXY_VIRTUAL_ENV" != "None" -a  "$_use_framework_galaxy" = "True" \
+         -a -f "$GALAXY_VIRTUAL_ENV/bin/activate" \
+         -a "`command -v python`" != "$GALAXY_VIRTUAL_ENV/bin/python" ]; then
+        . "$GALAXY_VIRTUAL_ENV/bin/activate"
+    fi
+}
+
 $headers
 $integrity_injection
 $slots_statement
 export GALAXY_SLOTS
+GALAXY_VIRTUAL_ENV="$galaxy_virtual_env"
```

Comment:
By moving this line here it will be possible to specify the Galaxy virtualenv on the cluster nodes inside the script specified by `environment_setup_file` or by adding `<env id="GALAXY_VIRTUAL_ENV">/PATH/TO/VIRTUALENV</env>` instead of having to use `<env file="/PATH/TO/VIRTUALENV/bin/activate" />` . Thanks @natefoo!

Reply:
Yeah - that is huge - it means someday we can conditionally not activate it (e.g. for ``profile>=18.01`` tools). We should probably scan the env for ``file=".../bin/activate"`` and issue a warning.

Reply:
That is where I am sitting I think.

Reply:
Not necessary IMO.

Reply:
Oh... well, it's there. Should I take it back out?

Reply:
It seems to unnecessarily complicate the code, but no strong feelings.